### PR TITLE
Multipart mime 3487 v33

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3487

Describe changes:
- convert HTTP to use new rust mime parser
- convert SMTP to use new rust mime parser

Follows #9963 with needed rebase after f5565f42e7ae993963d7135271813ff30780bc96 (which this PR fixed already)

### CODE REVIEW

File: rust/src/mime/mod.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (C) 2021 Open Information Security Foundation
+/* Copyright (C) 2022 Open Information Security Foundation
```
Comment (jasonish) [inline rust/src/mime/mod.rs:1]: Should this have been `2021-2022`?

**Review by jasonish — COMMENTED** [review]
Looking good on the Rust side. I haven't looked at everything as a whole yet though.  Is there some sort of mime torture test we can subject this to?

**Comment by catenacyber** [comment]
> Is there some sort of mime torture test we can subject this to?

I do not know any beyond what is in CI+QA

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on SURI_TLPW1_suri_time.

ERROR: QA failed on SURI_TLPW1_files_sha256.
  

Pipeline 18071

**Comment by catenacyber** [comment]
Rebased in https://github.com/OISF/suricata/pull/10376

**Pull request closed without merging**